sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/monitor/applicationinsights/spring/PropertiesPostProcessor.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.applicationinsights.spring;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+
+import java.util.Collections;
+import java.util.Map;
+
+class PropertiesPostProcessor implements EnvironmentPostProcessor, Ordered {
+    private static final Logger LOG = LoggerFactory.getLogger(PropertiesPostProcessor.class);
+
+    @Override
+    public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+
+        if (starterHasToBeDisabled(environment)) {
+            // To log once the logging system is initialized in the Spring Boot application
+            application.addInitializers(applicationContext -> LOG.warn("The spring-cloud-azure-starter-monitor Spring starter is disabled because the Application Insights Java agent is enabled."
```

Comment:
is it ok that `LOG` is statically initialized above already?

Reply:
The log message is printed in this way. I have included the log creation in the ApplicationContextInitializer now to prevent its usage in other code places.